docs/ising_qaoa.rst

```diff
@@ -0,0 +1,268 @@
+
+.. figure:: ising_qaoa/generalplan.png
+   :align: center
+   :figwidth: 100%
+
+The following sections give three concrete examples of how to use the
+Ising QAOA wrapper to approximate ground states of various Hamiltonians.
+
+.. _2d-checkerboard:
+
+2-local Checkerboard Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before we try to create a checkerboard pattern, let's quickly think about the interaction or coupling term.
+Consider two spins \\( \\sigma_{i} \\) and \\( \\sigma_{j} \\) and their 2-local interaction term \\( J_{i,j} \\):
+
+.. image:: ising_qaoa/simple_coupling.png
+   :align: center
+   :scale: 75%
+
+Suppose there are no biases on spins \\( i \\) and \\( j \\) and the coupling is \\( J_{i,j} = -1 \\).
+Which values of \\( \\sigma_{i} \\) and \\( \\sigma_{j} \\) minimize the energy? Both spins should have the same
+value, either +1 or -1, in order to get an overall energy of -1. Hence, a negative coupling term *correlates* spins!
+
+If the coupling is \\( J_{i,j} = +1 \\) the energy is minimized when the two spins have opposite values.
+Thus, a positive coupling *anti-correlates* spins!
+
+Now consider the following two-dimensional graph with spins \\( \\sigma_{0}, \\sigma_{1}, \\sigma_{2}, \\sigma_{3}\\):
+
+.. image:: ising_qaoa/2d_checkerboard.png
+   :align: center
+   :scale: 75%
+
+Let's define that we colour vertex \\( i \\) black if \\( \\sigma_{i} = -1 \\) and white if \\( \\sigma_{i} = +1 \\).
+The goal is to create a checkerboard pattern with the four vertices in the graph. There is various ways of defining
+\\( h \\) and \\( J \\) to achieve this result. There are two possible solutions to this problem:
+
+.. image:: ising_qaoa/2d_checkerboard_solutions.png
+   :align: center
+   :scale: 75%
+
+The example used in the :ref:`quickstart-example` is one way of doing it. However, it involved bias terms which
+strongly biased for solution nr. 2. This time we want don't care which solution we get as long as it is a valid
+solution. Hence, we don't use any bias terms and only anticorrelate each pair of neighbouring spins:
+
+.. code-block:: python
+
+    import pyquil.api as api
+    from grove.ising.ising_qaoa import ising_qaoa
+    qvm_connection = api.QVMConnection()
+
+    J = {(0, 1): 1, (0, 2): 1, (1, 3): 1, (2, 3): 1}
+    h = {}
+
+
+Given this Ising problem, we run the QAOA algorithm with two \\( \\beta \\) and two \\( \\gamma \\) parameters (``steps=2``).
+We run it ten times in order to collect some statistics:
+
+.. code-block:: python
+
+    steps = 2
+    runs = 10
+    stats = dict()
+    for _ in range(runs):
+        solution_string, ising_energy, _  = ising_qaoa(h=h, J=J, num_steps=steps)
+        if tuple(solution_string) in stats.keys():
+            stats[tuple(solution_string)] += 1
+        else:
+            stats[tuple(solution_string)] = 1
+    print(f'Solution statistics: {stats}')
+
+You should get the two possible solution strings \\( [-1, 1, 1, -1] \\) and \\( [1, -1, -1, 1] \\)
```

Comment:
I don't get this either.

Reply:
@ampolloreno This has also been resolved in commit [fc381d4](https://github.com/rigetticomputing/grove/pull/141/commits/fc381d4840a6325c2577ea5c48901b47769fe63a).

Reply:
:+1: